### PR TITLE
sql: make Eval return an error as well 

### PR DESCRIPTION
**Pull request by erizocosmico**

### CODE REVIEW

File: sql/expression/comparison.go [hunk 1]
```diff
@@ -68,26 +76,33 @@ func NewRegexp(left sql.Expression, right sql.Expression) *Regexp {
...
 
 	if !okl || !okr {
-		return re.ChildType.Compare(l, r) == 0
+		return re.ChildType.Compare(l, r) == 0, nil
 	}
 
 	reg, err := regexp.Compile(sr)
 	if err != nil {
-		return false
+		return false, nil
```
Comment (jfontan) [inline sql/expression/comparison.go]: should return err

Comment (erizocosmico) [reply sql/expression/comparison.go]: yes, good catch!

**Comment by erizocosmico** [comment]
Fixed @jfontan

File: sql/plan/filter.go [hunk 2]
```diff
@@ -55,10 +55,16 @@ type filterIter struct {
...
+			return nil, err
+		}
 
+		ok, err := i.f.expression.Eval(row)
 		if err != nil {
 			return nil, err
-		} else if i.f.expression.Eval(row) == true {
+		}
+
+		if ok == true {
```
Comment (ajnavarro) [inline sql/plan/filter.go]: with `if ok {` I think the result is the same :)

Comment (erizocosmico) [reply sql/plan/filter.go]: Nope: https://play.golang.org/p/XP_hLd18CJF

**Comment by erizocosmico** [comment]
Rebased

File: sql/plan/filter.go (hunk 2 above)
Comment (erizocosmico) [reply sql/plan/filter.go]: Nope: https://play.golang.org/p/XP_hLd18CJF

Comment (ajnavarro) [reply sql/plan/filter.go]: So, at least for me, have a `ok` variable name that usually means boolean feels a little bit strange for me. Maybe we should change the variable name for somethig less common, like `evaluated` ?

Comment (erizocosmico) [reply sql/plan/filter.go]: Yes, that makes sense. `ok` might induce to think it's a bool

**Comment by erizocosmico** [comment]
Fixed @ajnavarro

**Pull request merged**